Dockerfile

```diff
@@ -0,0 +1,47 @@
+# Development stage for installing dependencies
+FROM python:3.11.5-slim-bullseye as base
+
+# Set environment variables
+ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONUNBUFFERED 1
+
+# Install system dependencies
+RUN apt-get update && apt-get install -y curl
+
+# Install Poetry
+RUN curl -sSL https://install.python-poetry.org | python3 -
+
+# Create a virtual environment for dependencies
+RUN python3 -m venv /venv
+
+# Set the working directory
+WORKDIR /app
+
+# Copy only the dependency files for installation
+COPY pyproject.toml poetry.lock /app/
+
+# Install project dependencies using Poetry
+RUN /venv/bin/pip install poetry
+RUN /venv/bin/poetry install
+RUN /venv/bin/poetry add gunicorn
+
+# Copy the entire application code
+COPY . /app/
```

Comment:
If we're taking this approach of copying everything by default - and I quite like it as a simple approach - then we should definitely be adding a `dockerignore` file to avoid our `.env` file being copied in, but we may also want to ignore more things, e.g.:
```
.env
.venv
Dockerfile
docker-compose.yml
**/__pycache__
*.md

# any more?
```

Reply:
dockerignore file created.